O-Compiler/Source/Analyze/Lexical/Tokens/Token.cs

```diff
@@ -0,0 +1,69 @@
+﻿using OCompiler.Extensions;
+
+namespace OCompiler.Analyze.Lexical.Tokens
+{
+    abstract class Token
+    {
+        public long StartOffset { get; }
+        public int Length { get; }
+
+        public string Literal { get; }
+
+        public Token(long startOffset, string literal)
+        {
+            StartOffset = startOffset;
+            Length = literal.Length;
+            Literal = literal;
+        }
+
+        public static Token FromReserved(long position, Literals.ReservedLiteral literal) => literal switch
+        {
+            Literals.Boolean   => new BooleanLiteral(position, literal.Value),
+            Literals.Delimiter => new Delimiter(position, literal.Value),
+            Literals.Keyword   => new Keyword(position, literal.Value),
+            _ => throw new System.ArgumentException("Argument passed is not a reserved literal")
+        };
+
+        public static bool TryParse(long position, string literal, out Token token)
+        {
+            var reservedLiteral = Literals.ReservedLiteral.GetByValue(literal);
+            if (reservedLiteral == Literals.ReservedLiteral.Empty)
```

Comment:
Is it possible to make code less nested at the cost of more complex boolean expressions?

Reply:
~~at the cost of ternaries~~
will see

Reply:
Fixed with https://github.com/DOCtorActoAntohich/O-Compiler/pull/4/commits/ce64e8110617c32e2d17b69a0c637a7e74da7a3a